### PR TITLE
Add do_convert_rgb to vit

### PR DESCRIPTION
**Pull request by jp1924**
# What does this PR do?

While resolving issue #34301, discovered that ViT lacks do_rgb_convert.

Therefore, added do_rgb_convert to ViTImageProcessor.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@amyeroberts, @qubvel

### CODE REVIEW

File: src/transformers/models/vit/image_processing_vit.py [hunk 1]
```diff
@@ -82,6 +84,7 @@ def __init__(
         do_normalize: bool = True,
         image_mean: Optional[Union[float, List[float]]] = None,
         image_std: Optional[Union[float, List[float]]] = None,
+        do_convert_rgb: bool = None,
```
Comment (qubvel) [inline src/transformers/models/vit/image_processing_vit.py:87]:
```suggestion
        do_convert_rgb: Optional[bool] = None,
```

File: src/transformers/models/vit/image_processing_vit.py [hunk 2]
```diff
@@ -68,6 +68,8 @@ class ViTImageProcessor(BaseImageProcessor):
         image_std (`float` or `List[float]`, *optional*, defaults to `IMAGENET_STANDARD_STD`):
             Standard deviation to use if normalizing the image. This is a float or list of floats the length of the
             number of channels in the image. Can be overridden by the `image_std` parameter in the `preprocess` method.
+        do_convert_rgb (`bool`, *optional*, defaults to `self.do_convert_rgb`):
```
Comment (qubvel) [inline src/transformers/models/vit/image_processing_vit.py:71]:
```suggestion
        do_convert_rgb (`bool`, *optional*):
```

File: src/transformers/models/vit/image_processing_vit.py [hunk 3]
```diff
@@ -159,6 +163,7 @@ def preprocess(
         return_tensors: Optional[Union[str, TensorType]] = None,
         data_format: Union[str, ChannelDimension] = ChannelDimension.FIRST,
         input_data_format: Optional[Union[str, ChannelDimension]] = None,
+        do_convert_rgb: bool = None,
```
Comment (qubvel) [inline src/transformers/models/vit/image_processing_vit.py:166]:
```suggestion
        do_convert_rgb: Optional[bool] = None,
```

**Review by qubvel — COMMENTED** [review]
Hi @jp1924, thanks for adding this feature! To make it complete we also have to update `VitImageProcessorFast` to be compatible with this image processor. Please have a look if there is a required transform in torchvision or use Lambda transform. Thanks!

**Comment by jp1924** [comment]
Thanks for the advice @qubvel! That worked out well.

I wrote that type_hint with reference to siglip, should I modify siglip as well?

**Review by qubvel — COMMENTED** [review]
Hi @jp1924, thanks for the update! For fast image processor we also should add a transform that do RGB conversion, not only an argument.

**Comment by jp1924** [comment]
@qubvel 
Oh..... I thought I added `do_convert_rgb` to the fast processor, but I only added it to the arg??? haha😂
Thanks for letting me know!

**Review by qubvel — APPROVED** [review]
Thanks for the update! Looks good to me!

**Review by ArthurZucker — APPROVED** [review]
LGTM but as this was not requested before I am not entirely sure it will be used! 🤗

**Pull request merged**